src/test/scala/uk/gov/hmrc/perftests/disareturns/Util/JsonGenerator.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.perftests.disareturns.Util
+
+import com.typesafe.scalalogging.LazyLogging
+import play.api.libs.json.{JsError, JsObject, JsString, JsSuccess, Json}
+
+object JsonGenerator extends LazyLogging {
+  def generateSerializedNdjson(lines: Seq[String]): String = {
+    val sb = new StringBuilder
+
+    for ((line, i) <- lines.zipWithIndex) {
+      val trimmed = line.trim
+      if (trimmed.nonEmpty) {
+        try {
+          val jsValue = Json.parse(trimmed)
+          jsValue.validate[JsObject] match {
+            case JsSuccess(jsObject, _) =>
+              val updatedPayload =
+                jsObject +
+                  ("nino"                               -> JsString(RandomDataGenerator.nino())) +
+                  ("accountNumber"                      -> JsString(RandomDataGenerator.generateSTDCode())) +
+                  ("accountNumberOfTransferringAccount" -> JsString(RandomDataGenerator.generateOLDCode()))
+              sb.append(Json.stringify(updatedPayload)).append("\n")
+            case JsError(errors)        =>
+              logger.info(s"[WARN] Line ${i + 1}: Invalid JSON object, skipping. Errors: $errors")
+          }
+        } catch {
+          case ex: Exception =>
+            logger.info(s"[ERROR] Line ${i + 1}: Failed to parse JSON, skipping. Error: ${ex.getMessage}")
+        }
+      }
+    }
```

Comment:
All this seems to be doing is randomising 3 fields on an existing single line of text representing a json blob. That fixed json could be described as fixed values in a case class just as easily, with these three values added as parameters on a custom apply method. Pulling from the file and modifying the Json object looks like overcomplication to me.

Reply:
That's not using a case class with variables. If you've replaced this, then you should clean up the file.